backend/src/dao/TeamEventsDao.ts

```diff
@@ -1,107 +1,64 @@
         };
       })
     );
   }
 
-  static async getTeamEvent(uuid: string): Promise<TeamEvent> {
+  static async getTeamEvent(uuid: string): Promise<TeamEventInfo> {
     const eventDoc = teamEventsCollection.doc(uuid);
     const eventRef = await eventDoc.get();
     if (!eventRef.exists) throw new NotFoundError(`No form with uuid '${uuid}' found.`);
     const eventForm = eventRef.data();
     if (eventForm == null)
       throw new NotFoundError(`No form content in form with uuid '${uuid}' found.`);
     return {
-      ...eventForm,
-      requests: (await Promise.all(
-        eventForm.requests.map(async (ref) => ({
-          ...ref,
-          member: await getMemberFromDocumentReference(ref.member)
-        }))
-      )) as TeamEventAttendance[],
-      attendees: (await Promise.all(
-        eventForm.attendees.map(async (ref) => ({
-          ...ref,
-          member: await getMemberFromDocumentReference(ref.member)
-        }))
-      )) as TeamEventAttendance[]
+      ...eventForm
     };
   }
 
-  static async deleteTeamEvent(teamEvent: TeamEvent): Promise<void> {
+  static async deleteTeamEvent(teamEvent: TeamEventInfo): Promise<void> {
     const eventDoc = teamEventsCollection.doc(teamEvent.uuid);
     const eventRef = await eventDoc.get();
     if (!eventRef.exists) throw new NotFoundError(`No team event '${teamEvent.uuid}' exists.`);
     await eventDoc.delete();
   }
 
-  static async createTeamEvent(event: TeamEvent): Promise<TeamEvent> {
-    const teamEventRef: DBTeamEvent = {
+  static async createTeamEvent(event: TeamEventInfo): Promise<TeamEventInfo> {
+    const teamEventRef: TeamEventInfo = {
       uuid: event.uuid ? event.uuid : uuidv4(),
       name: event.name,
       date: event.date,
       numCredits: event.numCredits,
       hasHours: event.hasHours,
-      isCommunity: event.isCommunity,
-      requests: event.requests.map((req) => ({
-        ...req,
-        member: memberCollection.doc(req.member.email)
-      })),
-      attendees: event.attendees.map((att) => ({
-        ...att,
-        member: memberCollection.doc(att.member.email)
-      }))
+      isCommunity: event.isCommunity
     };
 
     await teamEventsCollection.doc(teamEventRef.uuid).set(teamEventRef);
     return event;
   }
 
-  static async updateTeamEvent(event: TeamEvent): Promise<TeamEvent> {
+  static async updateTeamEvent(event: TeamEventInfo): Promise<TeamEventInfo> {
     const eventDoc = teamEventsCollection.doc(event.uuid);
     const eventRef = await eventDoc.get();
     if (!eventRef.exists) throw new NotFoundError(`No team event '${event.uuid}' exists.`);
-    const teamEventRef: DBTeamEvent = {
-      ...event,
-      requests: event.requests.map((req) => ({
-        ...req,
-        member: memberCollection.doc(req.member.email)
-      })),
-      attendees: event.attendees.map((att) => ({
-        ...att,
-        member: memberCollection.doc(att.member.email)
-      }))
+    const teamEventRef: TeamEventInfo = {
```

Comment:
The usage of `teamEventRef` is not necessary here. You can just pass `event` into the `update(...)` method.